agent/src/test/java/com/blackducksoftware/integration/hub/teamcity/agent/scan/HubBuildProcessTest.java

```diff
@@ -90,7 +90,7 @@ public static void testStartup() throws Exception {
 
 		String workingDirPath = HubBuildProcessTest.class.getProtectionDomain().getCodeSource().getLocation().getPath();
```

Comment:
This process has come up in a few places...I keep changing it to avoid issues with multiple OS's to the following:
        URL url = Thread.currentThread().getContextClassLoader().getResource(classpathEntry);
        File file = new File(url.getFile());
        String absolutePath = file.getAbsolutePath();

where classpathEntry is the classpath url of whatever it is that you're looking for.